STYLE.md

```diff
@@ -0,0 +1,270 @@
+
+```
+resource "aws_iam_role" "iam_role" {
+  name = "${var.resource_name}-role"
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+```
+
+There should be two blank lines between resource definitions.
+
+```hcl
+// bad
+variable "vpc_id" {}
+
+variable "vpc_id" {
+  description = "The VPC this security group will go in"
+  type        = "string"
+}
+
+// goo
+variable "vpc_id" {
+  description = "The VPC this security group will go in"
+  type        = "string"
+}
+
+
+variable "public_subnet_ids" {
+  description = "List of public subnet IDs"
+  type        = "list"
+  default     = []
+}
+```
+
+
+
+### Resource Block Alignment
+
+Parameter definitions in a resource block should be aligned. The `terraform fmt` command can do this for you.
+
+```hcl
+// bad
+resource "aws_security_group" "main" {
+    name = "${var.name}"
+    description = "Security Group ${var.name}"
+    vpc_id = "${var.vpc_id}"
+    tags {
+        Name = "${var.name}"
+    }
+}
+
+// good
+resource "aws_security_group" "main" {
+  name        = "${var.name}"
+  description = "Security Group ${var.name}"
+  vpc_id      = "${var.vpc_id}"
+  tags {
+    Name = "${var.name}"
+  }
+}
+```
+### Variables
+
+Variables should be provided in a `variables.tf` file at the root of your module.
+
+A description and type should be provided for each declared variable. When it makes sense, please provide a reasonable default value.
+
+## Outputs
+
+Outputs should be provided in an `outputs.tf` file at the root of your module.
+
+Attributes exported by your modules should follow the same descriptive format as variable names.
+
+## Comments
+
+Only comment what is necessary.
+
+Single line comments: `#` or `//`.
+
+Multi-line comments: `/*` followed by `*/`.
+
+When commenting use two "//" and a space in front of the comment.
+
+```
+// CREATE ELK IAM ROLE 
+...
+```
+
+## Naming Conventions
+
+### File Names
+
+Create a separate resource file for each type of AWS resource. Similar resources should be defined in the same file and named accordingly.
+
+```
+ami.tf
+autoscaling_group.tf
+cloudwatch.tf
+iam.tf
+launch_configuration.tf
+providers.tf
+s3.tf
+security_groups.tf
+sns.tf
+sqs.tf
+user_data.sh
+variables.tf
+```
+
+### Parameter, Meta-parameter and Variable Naming
+
+ Only use an underscore (`_`) when naming Terraform resources like TYPE/NAME parameters and variables.
+
+ ```
+resource "aws_security_group" "security_group" {
+...
+ ```
+
+Variable names should be descriptive. Avoid abbreviations when reasonable.  Verbosity is a good thing.
+
+```hcl
+// bad
+variable "pub_sub_id" {}
+
+// good
+variable "public_subnet_id" {
+  // definition
+}
+```
+
+Boolean variables should follow the idiomatic pattern `is_<condition>`
```

Comment:
I think we should prefer to follow the variable name from the underlying resource as far as possible, as it makes it easier to know which variable maps to what between the module and the resources in it.